File: lua/LeetNeoCode/dependencies/LCIncludes.cs
// LeetCode common data structures for C#

using System;
using System.Collections.Generic;

// Definition for singly-linked list
public class ListNode {
    public int val;
    public ListNode next;

    public ListNode(int val = 0, ListNode next = null) {
        this.val = val;
        this.next = next;
    }
}

// Definition for a binary tree node
public class TreeNode {
    public int val;
    public TreeNode left;
    public TreeNode right;

    public TreeNode(int val = 0, TreeNode left = null, TreeNode right = null) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

// Definition for N-ary tree node
public class Node {
    public int val;
    public IList<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, IList<Node> _children) {
        val = _val;
        children = _children;
    }
}

File: lua/LeetNeoCode/dependencies/lc_includes.go
package main

// Definition for singly-linked list
type ListNode struct {
	Val  int
	Next *ListNode
}

// Definition for a binary tree node
type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

// Definition for n-ary tree node
type Node struct {
	Val      int
	Children []*Node
}
